### PR TITLE
pkg/actor(ticdc): reduce metrics overhead

### LINKED ISSUE #4584: Actor metrics consume lots of CPU

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4584

### What is changed and how it works?

* Reduce metrics overhead
* Add actor metrics to Grafana

```shell
# Master
go test -benchmem -run='^$' -bench '^(BenchmarkPollActor)$' github.com/pingcap/tiflow/pkg/actor

goos: linux
goarch: amd64
pkg: github.com/pingcap/tiflow/pkg/actor
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkPollActor/BenchmarkPollActor/1_actor(s)-40               639072              1763 ns/op              48 B/op          1 allocs/op
BenchmarkPollActor/BenchmarkPollActor/2_actor(s)-40               359673              3286 ns/op              96 B/op          2 allocs/op
BenchmarkPollActor/BenchmarkPollActor/4_actor(s)-40               240338              5152 ns/op             192 B/op          4 allocs/op
BenchmarkPollActor/BenchmarkPollActor/8_actor(s)-40               119098             11157 ns/op             384 B/op          8 allocs/op
BenchmarkPollActor/BenchmarkPollActor/16_actor(s)-40               56025             23284 ns/op             769 B/op         16 allocs/op
BenchmarkPollActor/BenchmarkPollActor/32_actor(s)-40               30073             43730 ns/op            1539 B/op         32 allocs/op
BenchmarkPollActor/BenchmarkPollActor/64_actor(s)-40               14853             89433 ns/op            3079 B/op         64 allocs/op
BenchmarkPollActor/BenchmarkPollActor/128_actor(s)-40               7292            159212 ns/op            6150 B/op        128 allocs/op
BenchmarkPollActor/BenchmarkPollActor/256_actor(s)-40               3501            317086 ns/op           12292 B/op        256 allocs/op
BenchmarkPollActor/BenchmarkPollActor/512_actor(s)-40               2161            613879 ns/op           24580 B/op        512 allocs/op
BenchmarkPollActor/BenchmarkPollActor/1024_actor(s)-40               991           1336090 ns/op           49185 B/op       1024 allocs/op
BenchmarkPollActor/BenchmarkPollActor/2048_actor(s)-40               518           2675111 ns/op           98335 B/op       2048 allocs/op
BenchmarkPollActor/BenchmarkPollActor/4096_actor(s)-40               268           4717841 ns/op          196966 B/op       4096 allocs/op
BenchmarkPollActor/BenchmarkPollActor/8192_actor(s)-40               126           9745629 ns/op          393234 B/op       8192 allocs/op
BenchmarkPollActor/BenchmarkPollActor/16384_actor(s)-40               66          19152056 ns/op          786499 B/op      16384 allocs/op
BenchmarkPollActor/BenchmarkPollActor/32768_actor(s)-40               28          38723166 ns/op         1573554 B/op      32774 allocs/op
PASS
ok      github.com/pingcap/tiflow/pkg/actor     25.878s

# This PR
go test -benchmem -run='^$' -bench '^(BenchmarkPollActor)$' github.com/pingcap/tiflow/pkg/actor 

goos: linux
goarch: amd64
pkg: github.com/pingcap/tiflow/pkg/actor
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkPollActor/BenchmarkPollActor/1_actor(s)-40               901522              1354 ns/op              48 B/op          1 allocs/op
BenchmarkPollActor/BenchmarkPollActor/2_actor(s)-40               356815              3095 ns/op              96 B/op          2 allocs/op
BenchmarkPollActor/BenchmarkPollActor/4_actor(s)-40               251820              4874 ns/op             192 B/op          4 allocs/op
BenchmarkPollActor/BenchmarkPollActor/8_actor(s)-40               138922              9189 ns/op             384 B/op          8 allocs/op
BenchmarkPollActor/BenchmarkPollActor/16_actor(s)-40               56138             18193 ns/op             769 B/op         16 allocs/op
BenchmarkPollActor/BenchmarkPollActor/32_actor(s)-40               32737             35717 ns/op            1539 B/op         32 allocs/op
BenchmarkPollActor/BenchmarkPollActor/64_actor(s)-40               16482             67022 ns/op            3077 B/op         64 allocs/op
BenchmarkPollActor/BenchmarkPollActor/128_actor(s)-40               8764            135848 ns/op            6149 B/op        128 allocs/op
BenchmarkPollActor/BenchmarkPollActor/256_actor(s)-40               4104            257987 ns/op           12292 B/op        256 allocs/op
BenchmarkPollActor/BenchmarkPollActor/512_actor(s)-40               2449            497959 ns/op           24577 B/op        512 allocs/op
BenchmarkPollActor/BenchmarkPollActor/1024_actor(s)-40              1270            988900 ns/op           49176 B/op       1024 allocs/op
BenchmarkPollActor/BenchmarkPollActor/2048_actor(s)-40               636           2091994 ns/op           98427 B/op       2048 allocs/op
BenchmarkPollActor/BenchmarkPollActor/4096_actor(s)-40               312           3863858 ns/op          196624 B/op       4096 allocs/op
BenchmarkPollActor/BenchmarkPollActor/8192_actor(s)-40               147           7903754 ns/op          393224 B/op       8192 allocs/op
BenchmarkPollActor/BenchmarkPollActor/16384_actor(s)-40               74          15789187 ns/op          789148 B/op      16386 allocs/op
BenchmarkPollActor/BenchmarkPollActor/32768_actor(s)-40               36          33688860 ns/op         1573152 B/op      32770 allocs/op
PASS
```

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/actor/system.go [hunk 1]
```diff
@@ -463,23 +492,26 @@ func (s *System) poll(ctx context.Context, id int) {
...
-			receiveDuration := time.Since(pollStartTime)
-			if receiveDuration > slowReceiveThreshold {
-				log.Warn("actor handle received messages too slow",
-					zap.Duration("duration", receiveDuration),
-					zap.Uint64("id", uint64(p.mb.ID())),
-					zap.String("name", s.name))
+			actorPollDuration := now().Sub(actorPollStartTime)
+			actorPollStartTime = approximateCurrentTime
+			if actorPollDuration > slowPollThreshold {
+				// Prometheus histogram is expensive, we only recrod slow poll.
```
Comment (sdojjy) [inline pkg/actor/system.go:506]: fix lint
```suggestion
				// Prometheus histogram is expensive, we only record slow poll.
```

File: pkg/actor/system.go [hunk 2]
```diff
@@ -414,18 +418,39 @@ func (s *System) Spawn(mb Mailbox, actor Actor) error {
...
 func (s *System) poll(ctx context.Context, id int) {
 	batchPBuf := make([]*proc, s.actorBatchSize)
 	batchMsgBuf := make([]message.Message, s.msgBatchSizePerActor)
 	rd := s.rd
 	rd.Lock()
 
-	startTime := time.Now()
+	// Approximate current time. It is updated when calling `now`.
+	var approximateCurrentTime time.Time
+	now := func() time.Time {
```
Comment (3AceShowHand) [inline pkg/actor/system.go:430]: why the `now` function? should `time.Now()` be enough?

**Review by 3AceShowHand — APPROVED** [review]

**Comment by asddongmen** [comment]
[](https://ci.pingcap.net/blue/organizations/jenkins/lint-check/detail/lint-check/945/pipeline/#step-55-log-141)[](https://ci.pingcap.net/blue/organizations/jenkins/lint-check/detail/lint-check/945/pipeline/#step-55-log-142)tools/bin/golangci-lint run --timeout 10m0s --skip-files kv_gen --skip-dirs dm,tests

[2022-02-15T03:07:10.169Z] pkg/actor/system.go:506:51: `recrod` is a misspelling of `record` (misspell)

[2022-02-15T03:07:10.169Z] 				// Prometheus histogram is expensive, we only recrod slow poll.

please fix this typo.

**Pull request merged**